enable/trait_defs/rgba_color_trait.py

```diff
@@ -7,58 +7,148 @@
 # is also available online at http://www.enthought.com/licenses/BSD.txt
 #
 # Thanks for using Enthought open source!
-""" Trait definition for an RGBA-based color, which is either:
+""" Trait definition for an RGBA-based color
 
-* A tuple of the form (*red*,*green*,*blue*,*alpha*), where each component is
-  in the range from 0.0 to 1.0
-* An integer which in hexadecimal is of the form 0xAARRGGBB, where AA is alpha,
-  RR is red, GG is green, and BB is blue.
+An RGBA-based color is a tuple of the form (*red*, *green*, *blue*, *alpha*),
+where each component is in the range from 0.0 to 1.0, which is the color
+representation used by Kiva for drawing.
+
+Two traits are made available, a casting trait which takes many different color
+representations and converts them to a tuple, and a mapped version of the trait
+which holds the unconverted value and has a shadow trait holding the RGBA tuple
+value.
 """
+import copy
 
 import numpy as np
 
-from pyface.color import Color
-from pyface.util.color_parser import ColorParseError, parse_text
+from pyface.color import Color as PyfaceColor
+from pyface.util.color_helpers import ints_to_channels
+from pyface.util.color_parser import color_table, parse_text
 from traits.api import TraitType
 from traits.trait_base import SequenceTypes
 
+# Placeholders for system- and toolkit-specific UI colors; the
+# toolkit-dependent code below will fill these with the appropriate
+# values.  These hardcoded defaults are for the Windows Classic
+# theme.
+color_table["syswindow"] = (0.83137, 0.81569, 0.78431, 1.0)
+
+
+def convert_to_color_tuple(value):
+    """Convert various color representations to Kiva color tuple.
+
+    This can accept integers of the form 0xRRGGBB, strings which
+    can be parsed by Pyface's color parser, (r, g, b) and (r, g, b, a) tuples
+    with either float 0.0 to 1.0 or int 0 to 255, Pyface Color objects, and
+    toolkit color objects.
+    """
+    if type(value) is int:
```

Comment:
Weaken this to `isinstance(value, int)`?

Reply:
Yep - that looks like code that was copied from the old version.
```suggestion
    if isinstance(value, int):
```